lib/iris/experimental/regrid.py

```diff
@@ -390,7 +390,7 @@ def _weighted_mean_with_mdtol(data, weights, axis=None, mdtol=0):
             weights_sum = weights.sum(axis=axis)
             frac_masked = 1 - np.true_divide(unmasked_weights_sum, weights_sum)
             mask_pt = frac_masked > mdtol
-            if np.any(mask_pt):
```

Comment:
A slight behaviour change between `numpy` versions `1.13.3` and `1.14.3` - you can no longer change the `mask` of a `MaskedConstant`. This makes total sense, and `numpy` have fixed quite rightly made this stricter.
```python
>>> import numpy as np
>>> np.__version__
'1.14.3'
>>> import numpy.ma as ma
>>> a = ma.array([1.0], mask=True)
>>> m = a[0]
>>> type(m)
<class 'numpy.ma.core.MaskedConstant'>
>>> m.mask |= True
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: output array is read-only
```
as opposed to
```python
>>> import numpy as np
>>> np.__version__
'1.13.3'
>>> import numpy.ma as ma
>>> a = ma.array([1.0], mask=True)
>>> m = a[0]
>>> type(m)
<class 'numpy.ma.core.MaskedConstant'>
>>> m.mask |= True
>>> m
masked
>>> m.mask |= False
>>> m
masked
```